src/rest-api/endpoints/management-endpoint/migration-endpoint.md

```diff
@@ -0,0 +1,215 @@
+---
+title: Migration Endpoint
+review:
+  comment: ''
+  date: '2023-03-16'
+  status: ok
+labels:
+  - http
+  - rest-api
+  - migration
+toc: true
+tree_item_index: 600
+---
+
+## Get All Migration Objects
+
+```
+GET /management/migration
+```
+
+### Response
+
+If successful, returns a JSON representation of all migration objects.
+
+### Status Codes
+
+- 200 _OK_ - Success.
+
+### Sample
+
+```curl
+curl -u Administrator:Administrator \
+http://localhost:8080/nuxeo/api/v1/management/migration
+```
+
+```json
+{
+  "entity-type": "migrations",
+  "entries": [
+    {
+      "entity-type": "migration",
+      "id": "trash-storage",
+      "description": "Migration of in the trash storage model",
+      "descriptionLabel": "migration.trash-storage",
+      "status": {
+        "state": "property",
+        "step": null,
+        "startTime": 0,
+        "pingTime": 0,
+        "progressMessage": null,
+        "progressNum": 0,
+        "progressTotal": 0,
+        "running": false
+      },
+      "steps": []
+    },
+    ...and so on...
+  ]
+}
+```
+
+## Get Migration Object
+
+```
+GET /management/migration/MIGRATION_ID
+```
+
+### Path Parameters
+
+| Parameter Name   | Type       | Description      |
+| ---------------- | ---------- | ---------------- |
+| **MIGRATION_ID** | **string** | The migration id.|
+
+If no `MIGRATION_ID` is passed, returns all migration objects, see [Get All Migrations](#get-all-migration-objects).
+
+### Response
+
+If successful, returns a JSON representation of the requested migration object.
+
+### Status Codes
+
+- 200 _OK_ - Success.
+- 404 _Not Found_ - Migration with the given `MIGRATION_ID` does not exist.
+
+### Sample
+
+```curl
+curl -u Administrator:Administrator \
+http://localhost:8080/nuxeo/api/v1/management/migration/trash-storage
+```
+
+```json
+{
+  "entity-type": "migration",
+  "id": "trash-storage",
+  "description": "Migration of in the trash storage model",
+  "descriptionLabel": "migration.trash-storage",
+  "status": {
+    "state": "property",
+    "step": null,
+    "startTime": 0,
+    "pingTime": 0,
+    "progressMessage": null,
+    "progressNum": 0,
+    "progressTotal": 0,
+    "running": false
+  },
+  "steps": []
+}
+```
+
+## Probe a Migration Object
+
+```
+POST /management/migration/MIGRATION_ID/probe
+```
+
+### Path Parameters
+
+| Parameter Name   | Type       | Description      |
+| ---------------- | ---------- | ---------------- |
+| **MIGRATION_ID** | **string** | The migration id.|
+
+### Response
+
+If successful, probes the current state of a migration by analyzing persistent data, and sets it as the new current state, then returns a JSON representation of the requested migration object.
```

Comment:
```suggestion
If successful, probes the current state of a migration by analyzing persistent data, sets it as the new current state, then returns a JSON representation of the requested migration object.
```
Though, not sure what "sets it as the new current state" means.

Reply:
Ok for removing the "and"
About the new current state. Here I'm quoting the Javadoc:
https://github.com/nuxeo/nuxeo-lts/blob/2021/modules/runtime/nuxeo-runtime-migration/src/main/java/org/nuxeo/runtime/migration/MigrationService.java#L236